sparkpost/base.py

```diff
@@ -22,6 +22,8 @@ def request(self, method, uri, **kwargs):
         response = requests.request(method, uri, headers=headers, **kwargs)
         if not response.ok:
             raise SparkPostAPIException(response)
+        if response.status_code == 204:
```

Comment:
We can handle this smarter later once the APIs all return 204s for DELETEs, but for now we need to return true or false based on if the response status code is 204. With that said, I'm wondering if we just put this code into each resource's ``delete`` methods until we can do something more sane.